cpp/models/lct_secir/parameters_io.cpp

```diff
@@ -0,0 +1,275 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "lct_secir/parameters_io.h"
+#include "memilio/config.h"
+
+#ifdef MEMILIO_HAS_JSONCPP
+
+#include "lct_secir/infection_state.h"
+#include "lct_secir/parameters.h"
+#include "memilio/io/epi_data.h"
+#include "memilio/io/io.h"
+#include "memilio/utils/date.h"
+#include "memilio/utils/logging.h"
+#include "memilio/math/eigen.h"
+
+#include <string>
+
+namespace mio
+{
+namespace lsecir
+{
+
+IOResult<Eigen::VectorXd> get_initial_data_from_file(std::string const& path, Date date, InfectionState infectionState,
+                                                     Parameters&& parameters, ScalarType total_population,
+                                                     ScalarType scale_confirmed_cases)
+{
+    // Try to get rki data from path.
+    BOOST_OUTCOME_TRY(rki_data, mio::read_confirmed_cases_noage(path));
+    auto max_date_entry = std::max_element(rki_data.begin(), rki_data.end(), [](auto&& a, auto&& b) {
+        return a.date < b.date;
+    });
+    if (max_date_entry == rki_data.end()) {
+        log_error("RKI data file is empty.");
+        return failure(StatusCode::InvalidFileFormat, path + ", file is empty.");
+    }
+    auto max_date = max_date_entry->date;
+    if (max_date < date) {
+        log_error("Specified date does not exist in RKI data.");
+        return failure(StatusCode::OutOfRange, path + ", specified date does not exist in RKI data.");
+    }
+    // Compute initial values for all subcompartments.
+    Eigen::VectorXd init = Eigen::VectorXd::Zero(infectionState.get_count());
+    // Define variables for parameters that are often needed.
+    ScalarType timeExposed            = parameters.get<TimeExposed>();
+    ScalarType timeInfectedNoSymptoms = parameters.get<TimeInfectedNoSymptoms>();
+    ScalarType timeInfectedSymptoms   = parameters.get<TimeInfectedSymptoms>();
+    ScalarType timeInfectedSevere     = parameters.get<TimeInfectedSevere>();
+    ScalarType timeInfectedCritical   = parameters.get<TimeInfectedCritical>();
+    ScalarType scale_mu_CR            = 1 / (1 - parameters.get<RecoveredPerInfectedNoSymptoms>());
```

Comment:
I find the name  `scale_mu_CR` not very descriptive, maybe something like `scale_asymptomatic` says more about why this factor is needed.

Reply:
Yes, i like the idea